pkg/relay/ethereum/tx.go

```diff
@@ -504,107 +518,105 @@ func (iter *CallIter) Current() sdk.Msg {
 func (iter *CallIter) End() bool {
 	return len(iter.msgs) <= iter.cursor
 }
-func (iter *CallIter) Next() {
-	if !iter.End() {
-		iter.cursor += 1
+func (iter *CallIter) Next(n int) {
+	iter.cursor = min(len(iter.msgs), iter.cursor + n)
+}
+
+func (iter *CallIter) updateLoggerMessageInfo(logger *log.RelayLogger, from int, count int) {
```

Comment:
Being buildTx more common function, caller of buildTx need to add info of msgs in using buildTx to logger.
This is helper function for it.